content/courses/advanced-ada/parts/data_types/types.rst

```diff
@@ -6,13 +6,293 @@ Types
 ..
     TO BE DONE:
 
-    Objects
-    -------
+    Names
+    -----
 
     .. todo::
 
-        - Definition of an object in Ada
-        - Difference to object-oriented programming
+        - Simple definition of names in Ada
+
+    .. admonition:: In the Ada Reference Manual
+
+        - :arm22:`4.1 Name <4-1>`
+
+
+.. _Adv_Ada_Objects:
+
+Objects
+-------
+
+The term *object* may be misleading for readers that have a strong background
+in object-oriented programming. Moreover, its meaning can vary depending on the
+context. Therefore, it's important to define what we mean by *objects* when
+focusing on Ada programming.
+
+In computer science, the term :wikipedia:`object <Object_(computer_science)>`
+can refer to a piece of data stored in memory |mdash| but it can also refer to
+a table or a form in a database. Also, even when we define the term *object* as
+data in memory, we can still classify programming languages as
+:wikipedia:`object-based <Object-based_language>` or
+:wikipedia:`object-oriented <Object-oriented_programming>` languages.
+
+.. admonition:: Important
+
+    In object-oriented programming, an object belongs to a *class* of objects.
+    In Ada, this kind of objects are called *tagged* objects. Note, however,
```

Comment:
This is a little awkward in English, because "this kind" is singular and "objects" is plural. 
Maybe " In Ada, objects of this kind are called tagged objects."  ?

Reply:
Thanks, I've just corrected it.